partitura/performance.py

```diff
@@ -280,6 +284,96 @@ def adjust_offsets_w_sustain(
         note["sound_off"] = offset
 
 
+class PerformedNote(dict):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self["id"] = self.get("id", None)
+        self["pitch"] = self.get("pitch", self["midi_pitch"])
+        self["note_on"] = self.get("note_on", -1)
+        self["note_off"] = self.get("note_off", -1)
+        self["sound_off"] = self.get("sound_off", self["note_off"])
+        self["track"] = self.get("track", 0)
+        self["channel"] = self.get("channel", 1)
+        self["velocity"] = self.get("velocity", 60)
+        self.validate_values()
+        self._accepted_keys = ["id", "pitch", "note_on", "note_off", "velocity", "track", "channel", "sound_off"]
+        self.__setitem__ = self._setitem_new
+
+    def __repr__(self):
+        return f"PerformedNote: {self['id']}"
+
+    def __str__(self):
+        return f"PerformedNote: {self['id']}"
+
+    def __eq__(self, other):
```

Comment:
I think that is dangerous to just test for equality by just checking the id. In any case, we should check all other attributes.